rpc/src/rpc.rs

```diff
@@ -656,6 +638,8 @@ impl JsonRpcRequestProcessor {
             .into());
         }
 
+        let partitioned_epoch_reward_enabled = epoch_boundary_block.num_reward_partitions.is_some();
```

Comment:
i guess we could have been using this logic all along?

i wonder if it'd make more sense to rename the variable something like `block_has_partitioned_rewards`? the current one kinda smells like a feature gate temporary

Reply:
> i guess we could have been using this logic all along?

Yes. 🤦‍♀️ I just didn't recognize that and over-complicated things.

> i wonder if it'd make more sense to rename the variable something like block_has_partitioned_rewards? the current one kinda smells like a feature gate temporary

Good point. I pushed an update; I went with `epoch_has_partitioned_rewards` because having partitioned rewards is a thing that stretches over multiple blocks.